fix(Sombra): prevent reentering stealth

arvrianmaybe opened the pull request:
fix sombra stealth cause even after you got out stealth you still can continue anways. This should hopefully fixes it.

MaxwellJung commented on src/heroes/sombra/stealth.opy:
```diff
@@ -73,6 +74,7 @@ rule "[sombra/stealth.opy]: Exit Stealth when taking damage":
     @Hero sombra
     @Condition eventPlayer.is_using_stealth
 
+    hideStealthHud
```
```suggestion
    hideStealthHud()
```

MaxwellJung commented on src/heroes/sombra/hack.opy:
```diff
@@ -13,3 +13,16 @@ rule "[sombra/hack.opy]: Apply OW1 hack on top of OW2 hack":
+    wait(OW2_SOMBRA_HACK_CAST_TIME, Wait.ABORT_WHEN_FALSE)
+    eventPlayer.disallowButton(Button.SECONDARY_FIRE)
+    eventPlayer.allowButton(Button.SECONDARY_FIRE)
+    damage(eventPlayer, null, 0.01)
+    heal(eventPlayer, null, 0.01)
```
```suggestion
    damage(eventPlayer, null, MIN_DAMAGE)
    heal(eventPlayer, null, MIN_DAMAGE)
```

MaxwellJung replied:
damaging 0.01 will result in sombra grunting when taking damage

arvrianmaybe replied:
i suppose so but i think this should be the only way resetting it or something but yeah  i think i could also find another way to fix this too. other than not using damages.

arvrianmaybe replied:
i guess for now this is an okay solution???

MaxwellJung replied:
MIN_DAMAGE is equivalent to 0.00001 damage or something

it should function the same as applying 0.01 damage, but with the added benefit of not making the character to grunt

MaxwellJung commented on src/heroes/sombra/stealth.opy:
```diff
@@ -1,4 +1,4 @@
-#!mainFile "../../dev_main.opy"
+#!mainFile "../main.opy"
```
```suggestion
#!mainFile "../dev_main.opy"
```

arvrianmaybe replied:
this still compiles normally but when you compile directly in the file it doesnt. The reason to this is cause i simply used a different variation on the file on one of my gamemodes and it uses that instead of the usual ""../dev_main.opy"

MaxwellJung commented on src/heroes/sombra/stealth.opy:
```diff
@@ -50,8 +50,8 @@ def enterStealth():
     hideStealthHud()
+    exitStealth()
     eventPlayer.setInvisibility(Invis.NONE)
-    eventPlayer.setMoveSpeed(percent(OW1_SOMBRA_STEALTH_MOVEMENT_BONUS/OW2_SOMBRA_STEALTH_MOVEMENT_BONUS))
```
You need this line for the movement speed bonus during "fake" stealth

arvrianmaybe replied:
i remove it due to it sometimes when youre not in stealth it still applys the move speed. So hence thats why i remove it the default speed still applys when i removed it which is 5.50 and 4.95. ill try to find a way to keep that in while not completly removing it.

arvrianmaybe replied:
and also just in general the passive stealth barely applys (like half a second ish so i dont think it matters too much but yeah i get your point, the speed still kinda matters when getting out of stealth.)

MaxwellJung reviewed (commented):
Not sure if I'll approve this; I'll keep it open for now

arvrianmaybe commented:
i have updated it slightly to fix most of the things, i have removed on the subroutines cause i dont think is needed. I dont mind if you need to get it back due to whatever reason such as the "style guidelines"

MaxwellJung commented:
Something is critically wrong because when I tested, I can't transition from fake stealth to real stealth.

Sombra will go into fake stealth, then play the ow2 stealth animation. However, the damage tick gets applied when she plays the animation, which forces her to exit stealth and consume cooldown.

I'm going to mark this as draft because it doesn't appear ready
